optimum/onnxruntime/configuration.py

```diff
@@ -604,73 +608,248 @@ def tensorrt(
     disable_shape_inference: bool = False
 
+    def __post_init__(self):
+        def deprecate_renamed_attribute(old_name, new_name, mapping_func=None):
+            if getattr(self, old_name, None) is not None:
+                if mapping_func is None:
+
+                    def identity(x):
+                        return x
+
+                    mapping_func = identity
+                setattr(self, new_name, mapping_func(getattr(self, old_name)))
+                warnings.warn(
+                    f"{old_name} will be deprecated soon, use {new_name} instead, {new_name} is set to "
+                    f"{getattr(self, new_name)}.",
+                    FutureWarning,
+                )
+
+        deprecate_renamed_attribute(
+            "optimize_with_onnxruntime_only",
+            "enable_transformers_specific_optimizations",
+            mapping_func=lambda x: not x,
+        )
+
+        deprecate_renamed_attribute("disable_gelu", "disable_bias_gelu_fusion")
+        deprecate_renamed_attribute("disable_layer_norm", "disable_layer_norm_fusion")
+        deprecate_renamed_attribute("disable_attention", "disable_attention_fusion")
+        deprecate_renamed_attribute("disable_skip_layer_norm", "disable_skip_layer_norm_fusion")
+        deprecate_renamed_attribute("disable_bias_skip_layer_norm", "disable_bias_skip_layer_norm_fusion")
+        deprecate_renamed_attribute("disable_bias_gelu", "disable_bias_gelu_fusion")
+        deprecate_renamed_attribute("disable_embed_layer_norm", "disable_embed_layer_norm_fusion")
+
+    def create_fusion_options(self, model_type: str) -> FusionOptions:
+        class Box:
+            pass
+
+        args = Box()
+        args.model_type = model_type
+        attribute_map = {
+            "disable_gelu_fusion": "disable_gelu",
+            "disable_layer_norm_fusion": "disable_layer_norm",
+            "disable_attention_fusion": "disable_attention",
+            "disable_skip_layer_norm_fusion": "disable_skip_layer_norm",
+            "disable_bias_skip_layer_norm_fusion": "disable_bias_skip_layer_norm",
+            "disable_bias_gelu_fusion": "disable_bias_gelu",
+            "disable_embed_layer_norm_fusion": "disable_embed_layer_norm",
+        }
+        for attr_name, fusion_attr_name in attribute_map.items():
+            setattr(args, fusion_attr_name, getattr(self, attr_name))
+
+        for attr, value in self.__dict__.items():
+            if hasattr(args, attr):
+                continue
+            setattr(args, attr, value)
+
+        return FusionOptions.parse(args)
+
+
+class AutoOptimizationConfig:
+    """
+    Factory to create common `OptimizationConfig`.
+    """
+
+    _LEVELS = {
+        "O1": {
+            "optimization_level": 1,
+            "enable_transformers_specific_optimizations": False,
+        },
+        "O2": {
+            "optimization_level": 2,
+            "enable_transformers_specific_optimizations": True,
+            "enable_gelu_approximation": True,
```

Comment:
Yeah I don't know if `enable_gelu_approximation` makes sense in a medium config :thinking: 
I don't have a strong opinion on this since I don't really know to which extent it impacts the model behaviour.

Reply:
I just added another optimization level.